api/src/env.ts

```diff
@@ -123,10 +125,39 @@ function getEnv() {
 	return dotenv.parse(fs.readFileSync(configPath).toString());
 }
 
+function getVariableType(variable: string) {
+	return variable.split(':').slice(0, -1)[0];
+}
+
+function getEnvVariableValue(variableValue: string, variableType: string) {
+	return variableValue.split(`${variableType}:`)[1];
+}
+
+function getEnvironmentValueByType(envVariableString: string) {
+	const variableType = getVariableType(envVariableString);
+	const envVariableValue = getEnvVariableValue(envVariableString, variableType);
+
+	switch (variableType) {
+		case 'number':
+			return toNumber(envVariableValue);
+		case 'array':
+			return toArray(envVariableValue);
+		case 'regex':
+			return new RegExp(envVariableValue);
+		case 'string':
+			return envVariableValue;
+	}
+}
+
 function processValues(env: Record<string, any>) {
 	env = clone(env);
 
 	for (const [key, value] of Object.entries(env)) {
+		if (acceptableEnvTypes.some((envType) => value.includes(`${envType}:`))) {
+			env[key] = getEnvironmentValueByType(value);
+			break;
```

Comment:
This shouldn't `break` the whole loop, but rather `continue` to the next one 🙂 

```suggestion
			continue;
```

Reply:
For context, this `break`:

https://github.com/directus/directus/pull/4190/files#diff-08db4c2c874e5cb71cb4b6caf511f45f26a590a4592fd6d2ebb2a1f0d580a5f3R165

is to break out of the `switch`, not the loop 🙂